Sources/swift-format/Frontend/Frontend.swift

```diff
@@ -176,35 +177,80 @@ class Frontend {
     at configurationFileURL: URL?,
     orInferredFromSwiftFileAt swiftFileURL: URL?
   ) -> Configuration? {
-    // If an explicit configuration file path was given, try to load it and fail if it cannot be
-    // loaded. (Do not try to fall back to a path inferred from the source file path.)
+    // The actual config file that this frontend is going to use.
+    let configURL: URL?
+
+    // If the config path is provided, use that.
+    // Otherwise, try inferring config path based on the input .swift file path.
     if let configurationFileURL = configurationFileURL {
-      do {
-        return try configurationLoader.configuration(at: configurationFileURL)
-      } catch {
-        diagnosticsEngine.emitError("Unable to read configuration: \(error.localizedDescription)")
-        return nil
-      }
+      configURL = configurationFileURL
+    } else if let swiftFileURL = swiftFileURL {
+      configURL = Configuration.url(forConfigurationFileApplyingTo: swiftFileURL)
+    } else {
+
+      // If neither path was given (for example, formatting standard input with no assumed filename)
+      // or if there was no configuration found by inferring it from the source file path, return the
+      // default configuration.
+      return Configuration()
     }
+    
+    // Now that we have the configURL, we're going to load it and validate the rules in it.
+    do {
+      // Ok to force unwrap because the only branch where configURL would not be set returned early.
+      let configuration = try configurationLoader.configuration(at: configURL!)
+      try self.verifyConfigurationRules(for: configuration, at: configURL!)
 
-    // If no explicit configuration file path was given but a `.swift` source file path was given,
-    // then try to load the configuration by inferring it based on the source file path.
-    if let swiftFileURL = swiftFileURL {
-      do {
-        if let configuration = try configurationLoader.configuration(forSwiftFileAt: swiftFileURL) {
-          return configuration
-        }
-        // Fall through to the default return at the end of the function.
-      } catch {
+      return configuration
+    } catch {
+      if let swiftFileURL {
         diagnosticsEngine.emitError(
           "Unable to read configuration for \(swiftFileURL.path): \(error.localizedDescription)")
-        return nil
+      } else {
+        diagnosticsEngine.emitError("Unable to read configuration: \(error.localizedDescription)")
       }
+      return nil
+    }
+  }
+
+  /// Checks if all the rules in the given configuration are supported by the registry.
+  /// If there are any rules that are not recognized by this version of swift-format, they are emitted as a warning.
+  ///
+  /// - parameters:
+  ///   - configuration: `Configuration` to verify
+  ///   - configURL: Configuration file `URL` to use in `Diagnostic.Location`
+  private func verifyConfigurationRules(for configuration: Configuration, at configURL: URL) throws {
```

Comment:
I'm _meh_ on naming here. It does get the job done, but I don't think the name is 💯. 

`verify` implies that if the config is invalid, it will throw and the program will exit with an error.

Perhaps an explicit `checkForUnrecognizedRules(in: config, at: path)`?

Also: I don't think it `throws` anymore.

Reply:
`checkForUnrecognizedRules(in: config, at: path)` sounds good to me. And yeah, remove the `throws` if it's no longer necessary.